crates/interledger-settlement/src/api.rs

```diff
@@ -92,70 +102,148 @@ impl_web! {
                             ))
                     };
 
+                    // TODO Idempotency header!
                     store_clone.update_balance_for_incoming_settlement(account_id, amount)
                         .map_err(move |_| {
                             error!("Error updating balance of account: {} for incoming settlement of amount: {}", account_id, amount);
-                            Response::builder().status(500).body(()).unwrap()
+                            Response::builder().status(201).body(()).unwrap() // Request was sent, but SE operation have failed.
```

Comment:
We shouldn't ack this if we haven't been able to process it. In this case, we know the processing failed so we should definitely return an error that will cause the SE to retry

```suggestion
                            Response::builder().status(500).body(()).unwrap() // Request was received but we were unable to process it (the settlement engine should retry this request)
```

Reply:
Agreed, however: https://forum.interledger.org/t/settlement-architecture/545/22?u=gakonst



```
201:
          description: Acknowledgement the endpoint recieved the request. NOTE, for
            the Settlement Engine this does not mean settlement has taken place
```

Reply:
It should only respond with a 201 if it saved the request to the store such that it does not need to be retried again and will be processed later. If we got an error and don't retry it on our side, we definitely shouldn't respond with a 2xx code

Reply:
That was my bad, I thought this was a SE operation, while it is a store operation. Definitely a `500` here.